### DISCUSSION TITLE
All snippets missing after upgrade

### DISCUSSION THREAD

**Discussion by el2493** [massCodeIO/massCode#225] [Q&A] [answered]
I got an alert that an update was available. I'm on Mac, so upgraded 3.2.0. When dragging the .app to the /Applications folder it prompted whether I wanted to overwrite the existing .app or leave both versions. I had a moment where I considered leaving both, but assumed the snippets would be stored somewhere other than in the .app and that if I upgraded (overwrote) then the new version would be able to access the same snippets.

However, when I opened the new version there were no snippets. Is there any way to get the snippets back? This will be very annoying if I have to re-create all of them.

**Comment by antonreshetov** [comment, marked as answer]
This should not be, a simple update does not overwrite the database. The database is not stored in the application, but in the path you specified either default `<home dir>/massCode`. Open db.json in the editor and check for the necessary snippets.

**Reply by el2493** [reply]
I found in Preferences the default location for Storage is /Users/[my account]/massCode. I went there in Finder and I do see snippets.db there (last modified earlier today when I turned on my computer). So it seems like the file is still there, but the new version isn't reading it.

I opened db.json (like you recommended). I see it was modified around the time I upgrade massCode and I see:

{
"folders": [
{
"id": "mvKT5Ypi",
"name": "Default",
"defaultLanguage": "typescript",
"parentId": null,
"isOpen": false,
"isSystem": false,
"createdAt": 1660233692547,
"updatedAt": 1660233692547
}
],
"snippets": [],
"tags": []
}

**Reply by antonreshetov** [reply]
So let's clarify what version you updated from, if 2.0.0-beta.1 then we have bad news

**Reply by el2493** [reply]
I don't know what the previous version was, but I had probably installed it over a year ago.

**Reply by antonreshetov** [reply]
Then there's an explanation https://github.com/massCodeIO/massCode/discussions/3

**Reply by el2493** [reply]
Thanks so much, that did work! I had tried to import before but didn't understand how it worked (I thought I had to select a specific .db file and not the folder). I just did it and all snippets are back.

Thanks again for your quick responses!

**Comment by andercard0** [comment]
I have to say that this issue just happened to me as well after the new update. Luckly I have all my snippets db backed up in other folder. 
That said, I'll move my storage folder to another in order to avoid data loss.

In case this could be investigated my scenario is as follows:
Masscode install path:
C:\1\massCode\massCode

Masscode storage path:
C:\1\massCode\Storage

As said, I'll move the mentioned path and wait for the next update to see/confirm if this is yet a problem in a different path.

**Reply by el2493** [reply]
If you had the same issue as I did (running a very old version of MassCode), I don't think changing the path will make a difference and you'll need to do the migration process.

**Reply by andercard0** [reply]
> If you had the same issue as I did (running a very old version of MassCode), I don't think changing the path will make a difference and you'll need to do the migration process.

Hm.. well no, was running 3.3.0 now 3.4.0 at the moment. never used a old version of it.

**Comment by Faq** [comment]
Hmm strange updated from 1.x to 3.x but cant find any db.json file to import. Finder shows only newly generated. By any chance it was called differently or got overwritten somehow? NO spec location was set before, just default install.
There are "backups" folder  and in it several folders with old dates, each folder has these files:
 massocde.db
 snippets.db
 tags.db
 
 Any way to use those files?

**Comment by ondrovic** [comment]
I have this issue an updated from 3.10.0 to 3.11.0. I had my snippets stored on another drive but it had been working perfectly now no matter what I do does it work. I tried uninstall, clearing registry, appdata, everything. Even moved storage back to ~/masscode folder with no luck.

I have even tried downgrading back to 3.10.0 and 3.9.0 and same results

Opening dev tools I do see these errors

```
Failed to load resource: net::ERR_CONNECTION_TIMED_OUT
index.af7082e3.js:12 Uncaught (in promise) Error: Network Error
    at GL (index.af7082e3.js:12)
    at XMLHttpRequest.h.onerror (index.af7082e3.js:13)
GL @ index.af7082e3.js:12
h.onerror @ index.af7082e3.js:13
index.af7082e3.js:12 GET http://localhost:3033/snippets/embed-folder net::ERR_CONNECTION_REFUSED
(anonymous) @ index.af7082e3.js:12
me @ index.af7082e3.js:12
setTimeout (async)
UX @ index.af7082e3.js:12
r @ index.af7082e3.js:12
getSnippets @ index.af7082e3.js:13
(anonymous) @ index.af7082e3.js:8
E @ index.af7082e3.js:2082
Fp @ index.af7082e3.js:4
nd @ index.af7082e3.js:4
n @ index.af7082e3.js:4
index.af7082e3.js:4 TypeError: Cannot read properties of null (reading 'length')
    at index.af7082e3.js:2082
    at r (index.af7082e3.js:4)
    at u1 (index.af7082e3.js:4)
    at Proxy.<anonymous> (index.af7082e3.js:2082)
    at M7 (index.af7082e3.js:4)
    at t6.wt [as fn] (index.af7082e3.js:4)
    at t6.run (index.af7082e3.js:1)
    at Fp (index.af7082e3.js:4)
    at dk (index.af7082e3.js:4)
rK @ index.af7082e3.js:4
C4 @ index.af7082e3.js:4
M7 @ index.af7082e3.js:4
wt @ index.af7082e3.js:4
run @ index.af7082e3.js:1
Fp @ index.af7082e3.js:4
dk @ index.af7082e3.js:4
Promise.then (async)
ck @ index.af7082e3.js:4
YT @ index.af7082e3.js:4
(anonymous) @ index.af7082e3.js:4
KS @ index.af7082e3.js:1
Rf @ index.af7082e3.js:1
(anonymous) @ index.af7082e3.js:1
set value @ index.af7082e3.js:1
(anonymous) @ index.af7082e3.js:1
getSnippets @ index.af7082e3.js:13
await in getSnippets (async)
(anonymous) @ index.af7082e3.js:8
E @ index.af7082e3.js:2082
Fp @ index.af7082e3.js:4
nd @ index.af7082e3.js:4
n @ index.af7082e3.js:4
index.af7082e3.js:13 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'sort')
    at Q8 (index.af7082e3.js:13)
    at Proxy.getSnippets (index.af7082e3.js:13)
Q8 @ index.af7082e3.js:13
getSnippets @ index.af7082e3.js:13
index.af7082e3.js:13 POST https://m.masscode.io/api/event net::ERR_CONNECTION_TIMED_OUT
(anonymous) @ index.af7082e3.js:13
iN @ index.af7082e3.js:13
VZ @ index.af7082e3.js:13
s6.request @ index.af7082e3.js:13
s6.<computed> @ index.af7082e3.js:13
(anonymous) @ index.af7082e3.js:12
pa @ index.af7082e3.js:13
setup @ index.af7082e3.js:2082
Fp @ index.af7082e3.js:4
eQ @ index.af7082e3.js:4
qk @ index.af7082e3.js:4
_e @ index.af7082e3.js:4
q @ index.af7082e3.js:4
z @ index.af7082e3.js:4
wt @ index.af7082e3.js:4
run @ index.af7082e3.js:1
Fp @ index.af7082e3.js:4
dk @ index.af7082e3.js:4
Promise.then (async)
ck @ index.af7082e3.js:4
uk @ index.af7082e3.js:4
oK @ index.af7082e3.js:4
B @ index.af7082e3.js:4
KS @ index.af7082e3.js:1
Jv @ index.af7082e3.js:1
(anonymous) @ index.af7082e3.js:1
KS @ index.af7082e3.js:1
Jv @ index.af7082e3.js:1
set value @ index.af7082e3.js:1
Ae @ index.af7082e3.js:12
(anonymous) @ index.af7082e3.js:12
Promise.then (async)
de @ index.af7082e3.js:12
j @ index.af7082e3.js:12
(anonymous) @ index.af7082e3.js:2128
emit @ VM14 node:events:394
onMessage @ VM112 renderer_init:73
index.af7082e3.js:12 Uncaught (in promise) Error: Network Error
    at GL (index.af7082e3.js:12)
    at XMLHttpRequest.h.onerror (index.af7082e3.js:13)
GL @ index.af7082e3.js:12
h.onerror @ index.af7082e3.js:13
error (async)
(anonymous) @ index.af7082e3.js:13
iN @ index.af7082e3.js:13
VZ @ index.af7082e3.js:13
s6.request @ index.af7082e3.js:13
s6.<computed> @ index.af7082e3.js:13
(anonymous) @ index.af7082e3.js:12
pa @ index.af7082e3.js:13
setup @ index.af7082e3.js:2082
Fp @ index.af7082e3.js:4
eQ @ index.af7082e3.js:4
qk @ index.af7082e3.js:4
_e @ index.af7082e3.js:4
q @ index.af7082e3.js:4
z @ index.af7082e3.js:4
wt @ index.af7082e3.js:4
run @ index.af7082e3.js:1
Fp @ index.af7082e3.js:4
dk @ index.af7082e3.js:4
Promise.then (async)
ck @ index.af7082e3.js:4
uk @ index.af7082e3.js:4
oK @ index.af7082e3.js:4
B @ index.af7082e3.js:4
KS @ index.af7082e3.js:1
Jv @ index.af7082e3.js:1
(anonymous) @ index.af7082e3.js:1
KS @ index.af7082e3.js:1
Jv @ index.af7082e3.js:1
set value @ index.af7082e3.js:1
Ae @ index.af7082e3.js:12
(anonymous) @ index.af7082e3.js:12
Promise.then (async)
de @ index.af7082e3.js:12
j @ index.af7082e3.js:12
(anonymous) @ index.af7082e3.js:2128
emit @ VM14 node:events:394
onMessage @ VM112 renderer_init:73
index.af7082e3.js:12 GET http://localhost:3033/snippets/embed-folder net::ERR_CONNECTION_REFUSED
(anonymous) @ index.af7082e3.js:12
me @ index.af7082e3.js:12
setTimeout (async)
UX @ index.af7082e3.js:12
r @ index.af7082e3.js:12
getSnippets @ index.af7082e3.js:13
(anonymous) @ index.af7082e3.js:8
E @ index.af7082e3.js:2082
Fp @ index.af7082e3.js:4
nd @ index.af7082e3.js:4
n @ index.af7082e3.js:4
index.af7082e3.js:13 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'sort')
    at Q8 (index.af7082e3.js:13)
    at Proxy.getSnippets (index.af7082e3.js:13)
Q8 @ index.af7082e3.js:13
getSnippets @ index.af7082e3.js:13```

**Reply by ondrovic** [reply]
For anyone else facing this problem:

here is a quick go package to parse the results into at least human readable I wrote this morning, helpful if you still have access to your db.json file

[masscode-parser](https://gist.github.com/ondrovic/00d4541d8610969b6757e9794ad0298c)
